### ISSUE TITLE
Open for a input field

### ISSUE THREAD

**Issue by nasirkhan** [UniSharp/laravel-filemanager#37]
I wan to use this file manager with a input field. If i click on a button, the file browser should appear and i will select the file. This feature is available in CKFinder can you please tell me now can i do it here?

**Comment by g0110280** [comment]
We haven't support this feature yet.
Happy to hear this suggestion, and we might add this later.

**Comment by youchenlee** [comment]
The behavior is very similar to the `Browse Server` button on CKEditor integrated with laravel-filemanager.

![screen_shot_2016-02-26_at_13_26_32](https://cloud.githubusercontent.com/assets/181350/13343988/c2fbc616-dc8c-11e5-9748-1f558015c75e.png)

The way to open file-manager window is very straightforward. When user click the button, it should open the path `/laravel-filemanager?type=Images` in a new window.

The question now is: How do we trigger the callback to pass the selected file to the opener 
As in https://github.com/UniSharp/laravel-filemanager/blob/master/src/views/index.blade.php#L361 for ckeditor / tinymce opener

Any suggestions? @nasirkhan

**Comment by nasirkhan** [comment]
@g0110280, can we get a expected release date? I integrated this package but as this feature is not available or i might not find i am searching for alternatives. For me it is very urgent. I found one (http://laravel-filemanager.rhcloud.com/) but the integration is not as easy as this one. Need your assistance to get it done. 

@youchenlee, I mentioned above that it is very urgent for me and i searched for alternatives. http://laravel-filemanager.rhcloud.com/ have this feature but have less feature as this package. 
You are right that the file manager window can be opened easily and i tried the approach as like the `guillermomartinez/filemanager-laravel`. Then file-manager window was opened but i could not select the files. I am not sure how to change the mentioned `useFile` function. I need assistance on this.

**Comment by youchenlee** [comment]
This instruction might help: https://github.com/simogeo/Filemanager/wiki/How-to-use-the-filemanager-from-a-simple-textfield
I will revise the code when having time.
Any suggestions or pull requests are very welcome.

**Comment by nasirkhan** [comment]
I tried to follow it instruction but it was not successful. i use the script as follows. When i click on the button the file browser showed up and when i select any image, the browser closes. but the input box remains empty. 

Another issue is, when i tried to use this script along with the CKEditior at the same time it shows an error `TypeError: this._.filebrowserSe is undefined`

```javascript
<script type="text/javascript">
  // File Picker modification for FCK Editor v2.0 - www.fckeditor.net
 // by: Pete Forde <pete@unspace.ca> @ Unspace Interactive
 var urlobj;

 function BrowseServer(obj)
 {
      urlobj = obj;
      OpenServerBrowser(
      '/laravel-filemanager?type=Images&CKEditor=description&CKEditorFuncNum=1&langCode=en',
      screen.width * 0.7,
      screen.height * 0.7 ) ;
 }

 function OpenServerBrowser( url, width, height )
 {
      var iLeft = (screen.width - width) / 2 ;
      var iTop = (screen.height - height) / 2 ;
      var sOptions = "toolbar=no,status=no,resizable=yes,dependent=yes" ;
      sOptions += ",width=" + width ;
      sOptions += ",height=" + height ;
      sOptions += ",left=" + iLeft ;
      sOptions += ",top=" + iTop ;
      var oWindow = window.open( url, "BrowseWindow", sOptions ) ;
 }

 function SetUrl( url, width, height, alt )
 {
      document.getElementById(urlobj).value = url ;
      oWindow = null;
 }
 </script>
```

```html
<button type="button" onclick="BrowseServer('image');">Pick Image</button>
<input type="text" id="image"/>
```

**Comment by nasirkhan** [comment]
great, you have started to work on this feature. do you have any tentative date to release?

**Comment by youchenlee** [comment]
Hi @nasirkhan, this is the primary objective in the next milestone. The release will be within two weeks, depends on our workload. Thank you for your previous contribution, it should help.

**Comment by nasirkhan** [comment]
thank you all, for your efforts. It is an awesome package, and i think it was one of the few missing parts. Waiting for the next release.

**Comment by nasirkhan** [comment]
@youchenlee , any update on this issue? it would be great if you can priorities this for me. :smile:

**Comment by youchenlee** [comment]
Hi @nasirkhan, 
@amin101 has made some changes for independent usage.
The key one should be at https://github.com/amin101/laravel-filemanager/commit/a1b516a0660647306350b82f053b5144ca8e285a#diff-dcfeb24959cdb58ee19a23037e3d7472R491

And please also read the "independent-use" section of the README: 
https://github.com/amin101/laravel-filemanager/blob/bugfix/README.md#independent-use

We will merge the pull request later.

**Comment by nasirkhan** [comment]
the new code works fine. :+1: 

a few observations 
* need to work for files other than images
* a different example could be added with text input and simple button

**Comment by g0110280** [comment]
Implemented in [1.5.0](https://github.com/UniSharp/laravel-filemanager/releases/tag/1.5.0)
